app/Http/Controllers/RecipeController.php

```diff
@@ -60,281 +62,177 @@ public function create()
+            NotificationFacade::send(User::getAdmins()->get(), new RecipeCreated($recipe->title, Auth::user()));
         }
 
+        $this->flashSuccessMessage('Recipe updated successfully.');
 
+        return redirect()->route('recipe.show', $recipe);
     }
 
     /**
      * Remove the specified resource from storage.
      */
     public function destroy(Recipe $recipe)
     {
+        $this->authorize('delete', $recipe);
 
-        try {
-            $this->authorize('delete', $recipe);
-            $collections = $recipe->collections()->withCount("recipes")->get();
-            foreach ($collections as $collection) {
+        $recipe
+            ->collections()
+            ->withCount("recipes")
+            ->get()
+            ->each(function ($collection) {
                 if ($collection->recipes_count == 1) {
                     $collection->delete();
                 }
-            }
-            $recipe->delete();
-
-            return redirect()->back();
-        } catch (Exception $e) {
-            session()->flash('alert', [
-                'title' => 'Error!',
-                'message' => $e->getMessage(),
-                'type' => 'error'
-            ]);
-            return redirect()->back();
-        }
+            });
 
+        $recipe->delete();
+        $this->flashSuccessMessage('Recipe deleted successfully.');
+
+        return Inertia::location(URL::previous());
     }
 
-    /*
-    Makes a recipe favorite to a user
-   */
-    public function favorite(Recipe $recipe, Request $request)
+    /**
+     *Makes a recipe favorite to a user
+     */
+    public function favorite(Recipe $recipe)
     {
-        try {
-            $this->authorize('update', $recipe);
-            $recipe->is_favorite = !$recipe->is_favorite;
-            $recipe->save();
-            return redirect()->back();
-        } catch (Exception $e) {
-            session()->flash('alert', [
-                'title' => 'Error!',
-                'message' => $e->getMessage(),
-                'type' => 'error'
-            ]);
-            return redirect()->back();
-        }
+        $this->authorize('update', $recipe);
+        $recipe->is_favorite = !$recipe->is_favorite;
+        $recipe->save();
+
+        return redirect()->back();
     }
 
-    /*
-        Creates a review(rate the recipe)
-       */
+    /**
+     * Creates a review(rate the recipe)
+     */
     public function rate(Recipe $recipe, Request $request)
     {
-        try {
-            $this->authorize('view', $recipe);
-
-            $rating = $recipe->reviews()->where("user_id", Auth::user()->id)->first();
-
-            $request->request = $request->validate([
-                "rating" => ['required', 'integer', 'max:5', 'min:1'],
-                "msg" => ['required', 'string', 'max:500', 'min:1']
-            ],
-                [
-                    "rating.required" => "Rating is required!",
-                    "msg.required" => "Message is required!",
```

Comment:
Validation should be done on front-end , custom messages from back are not option here